core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -670,13 +671,15 @@ public PlanOptimizers(
                 // pushdown into the connectors. We invoke PredicatePushdown and PushPredicateIntoTableScan after this
                 // to leverage predicate pushdown on projected columns.
                 new StatsRecordingPlanOptimizer(optimizerStats, new PredicatePushDown(metadata, typeOperators, typeAnalyzer, true, false)),
-                simplifyOptimizer,  // Should be always run after PredicatePushDown
                 new IterativeOptimizer(
                         metadata,
                         ruleStats,
                         statsCalculator,
                         estimatedExchangesCostCalculator,
-                        ImmutableSet.of(new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))),
+                        ImmutableSet.<Rule<?>>builder()
+                                .addAll(simplifyOptimizerRules) // Should be always run after PredicatePushDown
+                                .add(new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))
```

Comment:
Could you just add `PushPredicateIntoTableScan` to keep it simple?

Reply:
no, i don't think so

Reply:
Why not>?

Reply:
`simplifyOptimizer` is invoked in multiple places, some not showing up in the diff. i didn't want to make bigger change out of this.

also, arguably `PushPredicateIntoTableScan` is not just "simplification of the plan", so i'd need to come up with a new name too

Reply:
I think it doesn't look good and adds will rise additional question marks for future readers. Maybe just merge the last iterative optimizers?

Reply:
anyways, it would be great to note somewhere (comment) why we need to run some of the rules after PPD.
It's totally not obvious why we need to run `PushPredicateIntoTableScan` after PPD